<a href="https://colab.research.google.com/github/zodongchan/zodongchan/blob/main/1%E1%84%8B%E1%85%B5%E1%86%AF%E1%84%8E%E1%85%A1%E1%84%80%E1%85%AA%E1%84%8C%E1%85%A6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

안녕하세요.<br><br>
수강생 여러분~! Chapter04 수업은 잘 들으셨나요?

Chapter04의 수업 내용을 바탕으로 아래 문제를 풀어주시면 됩니다.

<br>

[**답변 입력 방법**]

문제 아래에 있는 코드 입력창에 정답코드를 작성하신 후  __▷ 버튼을 클릭해 코드를 실행__ 해보세요<br>
<br>

-----

## 도전! 신비의 섬 탐험

![image.png](https://media.cnn.com/api/v1/images/stellar/prod/180219103122-zanzibar-and-its-islands-mnemba-a-view-from-the-sky-mnemba-island-lodge.jpg?q=w_1600,h_900,x_0,y_0,c_fill/h_447)

### [배경 이야기]

탐험가인 당신은 신비의 섬에 도착했습니다.


이 섬에는 숨겨진 보물이 있으며, 이를 찾기 위해 여러 도전 과제를 해결해야 합니다.


섬에는 알 수 없는 외계인이 당신을 쫓아오고 있어 모든 도전을 신속하게 해결하고 섬을 탈출해야 합니다.

=================================================================================================================================

### [문제 풀이 팁]
너무 어렵게 생각하지 마세요!
지금은 배우는 중이니까 틀려도 괜찮아요.

<br>

문제를 풀 때는 아래 순서대로 천천히 해보세요 !

1. 문제를 __잘__ 읽어주세요!
    - 어떤 결과값을 요구하는지, 어떤 데이터를 주고 있는지 파악합니다.
2.  어떻게 구할 수 있을지 __'생각'__ 해봅니다!
    - 결과값을 구하기 위한 로직을 생각해봅니다.
3. 파이썬 문법에 맞춰서 __'하나씩'__ 코드로 옮겨주세요!
    - 물론 틀려도 괜찮습니다. 기억안나도 괜찮아요! 그래도 시도해보세요!

<br>

머릿속으로만 고민하지 말고, 직접 종이에 적거나 손으로 그려보는 것도 큰 도움이 됩니다! <br>
화이팅😃  

<br>

=================================================================================================================================

### [(심화 ver.) 예습을 해보고 싶다면?]

신비의 섬을 나가기 위해 필요한 비밀번호를 **변수**에 담아 놓는다면 번거로운 계산 과정이 줄어듭니다.

<br>

- 🐣 : 변수가 뭔지 잘 모르겠어요 !!!!!!!!!!!!<br>
💁 : __Chapter02 강의 예습__ 혹은 __검색__을 통해 간단한 개념을 파악할 수 있습니다!


<br>


##### __<신비의 섬 탐험> 문제는 성장을 위해 난이도가 살짝 높게 작성되었기 때문에 어려울 수 있지만 도전해보시길 바랍니다 :)__



<br>
<hr/>


__[문제 1: 입구의 수수께끼]__

<br>

섬의 입구에 도착한 당신은 좋지 않은 느낌을 받습니다. 무언가 비릿한 냄새가 여기저기에서 납니다.

아마도 나보다 앞서 도착한 도전자들의 시체가 썩는 냄새인 것 같습니다.

지금...누군가 당신을 지켜보고 있습니다. 한시라도 빠르게 이 섬을 탈출해야 합니다.

우선 이 거대한 돌문을 열고 섬의 더 깊숙한 곳으로 들어가야 합니다. 돌문을 열려면 두 개의 열쇠 숫자를 더해야 합니다.

열쇠 숫자는 14와 29입니다.

__돌문을 여는 숫자__ 를 구해주세요.

In [ ]:
비밀번호 = 14 + 29
print(비밀번호)

43


<hr/>
<br>

__[문제 2: 첫 번째 강의 다리]__

<br>



돌문을 지나자 거친 강물이 흐르는 강이 나타났습니다.

아주 낡은 나무 다리가 힘겹게 메달려 있습니다.

당장이라도 무너질 것 같지만 이 다리 말고는 반대편으로 갈 수 있는 길이 보이지 않습니다.

무언가 불길한 기운이 계속 가까워지는 것만 같아 당신 빠르게 다리를 건너야 합니다.

다리는 50kg 이상 버티지 못하기 때문에 68kg인 당신은 무게를 줄여야 합니다.

방금 전 돌문을 통과하면서 얻은 마법 룬을 이용해 무게를 줄일 수 있습니다.

다리를 건너기 위해 줄여야 할 __최소 무게__ 를 구해주세요. (이상, 이하, 초과, 미만은 신경쓰지 않습니다!)


<br>

In [ ]:
줄일몸무게 = 68 - 50
print(줄일몸무게)

18



<hr/>
<br>

__[문제 3: 숲 속의 고목나무]__

<br>



첫 번째 강을 건너자 으스스한 숲 속에 들어섰습니다.

귀를 찢는 울음소리가 숲 속에 울려 퍼집니다. 당신은 바로 오래된 고목나무 숲으로 몸을 숨깁니다.

숲 깊이 들어가자 고목 나무에 걸린 현판이 보입니다.

"나무의 나이를 구하고 연못으로 가라"

나무의 나이를 구하기 위해서는 4년마다 한 번의 나이테가 생깁니다.

돌문을 열때 사용한 숫자를 나이테 개수로 사용하여 __나무의 나이__ 를 구하세요.



__Tips__
- 문제 1의 결과값을 참고하세요 !

In [ ]:
나무나이 = 비밀번호 * 4
print(나무나이)

172


<hr/><br>

__[문제 4: 숲 속의 마법 연못]__

<br>



고목나무를 지나자 신비로운 마법 연못이 나타났습니다.

연못 속에는 불가사의한 에너지가 흐르고 있습니다.

이 에너지를 이용하면 보물을 찾는 데 필요한 단서를 얻을 수 있습니다.

연못의 깊이를 측정하려면 나무의 나이를 이용해야 합니다.

나무의 나이가 172살이고, 연못의 깊이는 나무 나이의 1/4입니다.

__연못의 깊이__ 를 구하세요.

<br>



In [ ]:
연못_깊이 = tree / 4
print(연못_깊이)

43.0


<hr/><br>

__[문제 5: 사원의 입구]__

<br>



연못을 지나니 고대 사원이 나타났습니다. 사원의 입구에는 비밀번호를 입력해야 합니다.

비밀번호는 3자리 수로, 첫 번째 자리는 5, 두 번째 자리는 돌문을 여는 숫자의 십의 자리 수, 세 번째 자리는 돌문을 여는 숫자의 일의 자리 수의 3분의 1입니다.

__비밀번호__ 를 구하세요.

<br>

In [ ]:

첫번째_자리 = 5
두번째_자리 = 비밀번호 // 10  # 십의 자리 수
세번째_자리 = (비밀번호 % 10) // 3 # 일의 자리 수의 3분의 1

입구_비밀번호 = 첫번째_자리, 두번째_자리, 세번째_자리
print(입구_비밀번호)

(5, 4, 1)


<hr/><br>

__[문제 6: 비밀의 동굴]__

<br>

사원 내부에 비밀의 동굴로 이어지는 통로가 있습니다.

무언가 쿵 하고 넘어지는 소리가 들렸습니다. 신경을 곤두세우는 까마귀 소리가 여기저기서 울려 퍼집니다.

주변을 멤돌던 존재가 사원에 진입한것 같습니다.

어서 동굴을 벗어나야할 것 같습니다.

동굴을 열려면 암호를 풀어야 합니다. 암호는 돌문을 열때 사용한 숫자의 제곱근과 같습니다.

__암호__ 를 구하세요.

<br>

In [ ]:
동굴_비밀번호 = int(비밀번호 ** 0.5)
print(동굴_비밀번호)

6


<hr/><br>

__[문제 7: 해변의 파도]__

<br>


비밀의 동굴을 지나 해변에 도착했습니다.

동굴에서 빠져나오진 못한 존재는 엄청난 촉수를 가진 거대 외계인으로 보입니다. 외계인은 당신을 보며 군침을 흘리고 있습니다. 빠르게 이곳을 탈출해 섬의 반대편으로 도망가야 합니다.

배를 타고 빠르게 벗어나기 위해 우선 파도를 파악해야합니다.

파도가 일정한 간격으로 오고 있음을 발견했습니다.

5분 동안 20번의 파도가 왔다면, 파도 한 번이 오는 데 걸리는 __평균 시간__ 을 구하세요.

<br>

In [ ]:
초당파도 = 5 * 60
파도 = 초당파도 / 20

print(f'{파도} 초당 1번')

15.0 초당 1번


<hr/><br>

__[문제 8: 폭포의 높이]__

<br>

섬의 반대편의 무사히 도착했습니다.

거대한 폭포가 당신의 길을 가로 막고 있습니다.

자세히 보니 폭포 위쪽에 반짝이는 무언가가 있습니다. 당신이 찾던 보물일 수 있습니다.

마법 룬을 이용해 폭포의 높이를 구하고 폭포의 위로 올라갑니다.

물방울이 떨어져 땅에 닿는 데 걸리는 시간은 2초입니다. 중력가속도는 9.8m/s²입니다.

돌문을 열때 사용한 숫자를 이용해 __폭포의 높이__ 를 구하세요. (단, 공기저항은 무시합니다.)

<br>


__Tips__
- 높이 = 0.5 * 중력가속도 * (시간 + 돌문의 숫자의 마지막 자리 수)²

<br>

In [ ]:
높이 = 0.5 * 9.8*(2 + int(비밀번호)%10)**2
print(높이)

122.50000000000001


<hr/>
<br>

__[최종 문제: 탈출 비밀번호]__

<br>

보물을 찾았습니다.


"쿠우에에이이이이이이.."


에일리언이 당신을 쫓아왔습니다. 당신은 곧 에일리언의 좋은 영양소가 될 것 같습니다.

당신은 지금 정신을 차리고 앞에 보이는 에일리언의 우주선을 뻇어 섬을 탈출해야합니다.

우주선에 타기 위해서는 비밀번호를 입력해야 합니다.

비밀번호는 현재 좌표의 첫 번째 숫자와 세 번째 숫자의 차와 같습니다.

__비밀번호__ 를 알아내세요!

<br>

현재 좌표 : 32, 43, 128

<br>

In [ ]:
현재_좌표 = 32, 43, 128
우주선_비밀번호 = 현재_좌표[2] - 현재_좌표[0]
print(우주선_비밀번호)

96


<hr/><br>

당신은 생존했습니다 !!!!!!!!!!

생존한 당신은 보물 상자를 열어봅니다.



![image.png](https://t1.daumcdn.net/brunch/service/user/f2nO/image/5kAK9UfcMgaAGLdejuEuAoS0Hos.jpg)

-----
__[편지]__ <br>
>  <br> 오늘도 고생하셨습니다.<br><br> 내일도 과제 있습니다. 여러분 : )<br><br>-파머-<br>


---